src/apollo/queries.ts

```diff
@@ -242,6 +248,38 @@ export const ENS_REGISTRATIONS = gql`
   }
 `;
 
+export type EnsGetRegistrationData = {
+  registration: {
+    id: string;
+    registrationDate: number;
+    expiryDate: number;
+    registrant: {
+      id: string;
+    };
+  };
+};
+
+export const ENS_GET_REGISTRATION = gql`
+  query getRegistration($id: ID!) {
```

Comment:
The new queries here I couldn't directly retrieve from the smart contract.

Reply:
wondering if we should merge this with https://github.com/rainbow-me/rainbow/pull/2997/files/52fe11345822d35661a51b248eb92a2e7d2d6c43#diff-c196b0d95a4c3d59bf1d172d6819e9d4881b07792e69c4d87a20b80d2bd67fdaR241

Reply:
export const ENS_REGISTRATIONS = gql`